chainstate/test-framework/src/block_builder.rs

```diff
@@ -323,14 +324,17 @@ impl<'f> BlockBuilder<'f> {
     }
 
     /// Builds a block without processing it.
-    pub fn build(self) -> Block {
-        self.build_impl().0
+    pub fn build(self, rng: &mut (impl Rng + CryptoRng)) -> Block {
```

Comment:
I'd consider moving this to the constructor and storing a ref inside the builder. This would allow chaining methods that require `rng`. Otherwise rust will complain about multiple mutable references.

Reply:
Storing a mutable reference is a big pain. Not only will this start requiring life time generic parameter, but also every test then will require multiple rngs. Now only some do due to the use of closures with mutable rng references.